pkg/operator/revisioncontroller/revision_controller.go

```diff
@@ -353,42 +353,59 @@ func (c RevisionController) sync(ctx context.Context, syncCtx factory.SyncContex
 		return nil
 	}
 
-	if shouldCreateNewRevision, err := c.revisionPrecondition(ctx); err != nil || !shouldCreateNewRevision {
-		return err
-	}
-
 	// If the operator status's latest available revision is not the same as the observed latest revision, update the operator.
+	// This needs to be done even if the revision precondition is not required because it ensures our operator status is
```

Comment:
Okay I think i understand it now. the CM is created by the createNewRevision which is used by the `getLatestAvailableRevision` method below. 

which means that the revision was created thus we should set it, right ?

Reply:
> Okay I think i understand it now. the CM is created by the createNewRevision which is used by the `getLatestAvailableRevision` method below.
> 
> which means that the revision was created thus we should set it, right ?

What is CM in that sentence?

If a revision exists and is ready (adding that code now), then always set the value.  This catches cases were a previous update failed.